packages/eslint-plugin/src/rules/no-useless-default-assignment.ts

```diff
@@ -236,25 +247,63 @@ export default createRule<[], MessageId>({
       }
     }
 
-    function reportUselessDefault(
+    function reportUselessDefaultAssignment(
       node: TSESTree.AssignmentPattern,
       type: 'parameter' | 'property',
-      messageId: MessageId,
     ): void {
       context.report({
         node: node.right,
-        messageId,
+        messageId: 'uselessDefaultAssignment',
         data: { type },
-        fix(fixer) {
-          // Remove from before the = to the end of the default value
-          // Find the start position (including whitespace before =)
-          const start = node.left.range[1];
-          const end = node.range[1];
-          return fixer.removeRange([start, end]);
+        fix: fixer => removeDefault(fixer, node),
+      });
+    }
+
+    function reportUselessUndefined(
+      node: TSESTree.AssignmentPattern,
+      type: 'parameter' | 'property',
+    ): void {
+      context.report({
+        node: node.right,
+        messageId: 'uselessUndefined',
+        data: { type },
+        fix: fixer => removeDefault(fixer, node),
+      });
+    }
+
+    function reportPreferOptionalSyntax(
+      node: TSESTree.AssignmentPattern,
+    ): void {
+      context.report({
+        node: node.right,
+        messageId: 'preferOptionalSyntax',
+        fix: fixer => {
+          const fixes = [removeDefault(fixer, node)];
+
+          const { left } = node;
+          if (left.type === AST_NODE_TYPES.Identifier) {
+            fixes.push(
+              fixer.insertTextAfterRange(
+                [left.range[0], left.range[0] + left.name.length],
+                '?',
+              ),
+            );
+          }
+
+          return fixes;
         },
```

Comment:
nit - you can use the generator form for this to save a few lines.

```suggestion
        *fix(fixer) {
          yield removeDefault(fixer, node);

          const { left } = node;
          if (left.type === AST_NODE_TYPES.Identifier) {
            yield fixer.insertTextAfterRange(
              [left.range[0], left.range[0] + left.name.length],
              '?',
            );
          }
        },
```

Reply:
Oh that's cool. I never really used generators before, but this looks handy.